AdyenComponents/ACH Direct Debit/ACHDirectDebitComponent.swift

```diff
@@ -0,0 +1,283 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+#if canImport(AdyenEncryption)
+    import AdyenEncryption
+#endif
+import UIKit
+
+/// A component that provides a form for ACH Direct Debit payment.
+public final class ACHDirectDebitComponent: PaymentComponent, PresentableComponent, Localizable, LoadingComponent, PublicKeyConsumer {
+    
+    private enum ViewIdentifier {
+        static let headerItem = "headerItem"
+        static let holderNameItem = "holderNameItem"
+        static let bankAccountNumberItem = "bankAccountNumberItem"
+        static let bankRoutingNumberItem = "bankRoutingNumberItem"
+        static let billingAddressItem = "billingAddressItem"
+        static let payButtonItem = "payButtonItem"
+    }
+    
+    /// :nodoc:
+    public let apiContext: APIContext
+    
+    /// :nodoc:
+    public var paymentMethod: PaymentMethod {
+        achDirectDebitPaymentMethod
+    }
+
+    /// :nodoc:
+    public weak var delegate: PaymentComponentDelegate?
+    
+    /// Component configuration
+    public let configuration: Configuration
+
+    /// :nodoc:
+    public lazy var viewController: UIViewController = SecuredViewController(child: formViewController, style: style)
+
+    /// :nodoc:
+    public var localizationParameters: LocalizationParameters?
+
+    /// Describes the component's UI style.
+    public let style: FormComponentStyle
+
+    /// :nodoc:
+    public let requiresModalPresentation: Bool = true
+
+    /// :nodoc:
+    public let shopperInformation: PrefilledShopperInformation?
+    
+    /// :nodoc:
+    public var publicKeyProvider: AnyPublicKeyProvider
```

Comment:
can this be let or private(set)? also the in the consumer protocol it doesn't have to be set only get is enough